File: Sem- 1/C/Function/factorial_recursion.c
#include<stdio.h>
int factorial(int n);
void main(){
	printf("Factorial of a Number using Recursion\n");
	int n;
	printf("Enter any number: ");
	scanf("%d",&n);
	printf("Facorial of %d = %d",n,factorial(n));
}
int factorial(int n){
	if(n>1){
		return n*factorial(n-1);
	}
	else{
		return 1;
	}
}

File: Sem- 3/Data Structure/Array/FactorialRecursion.java
import java.util.Scanner;
public class FactorialRecursion{
	public static void main(String[] args){
		Scanner sc=new Scanner(System.in);
		System.out.println("Facorial of a Number using Recursion");
		System.out.print("Enter any number: ");
		int n=sc.nextInt();
		if(n<0){
			System.out.println("Not Defined");
		}else{
			int ans=factorial(n);
			System.out.println(n+"! = "+ans);
		}
	}
	public static int factorial(int n){
		if(n==0||n==1){
			return 1;
		}else{
			return n*factorial(n-1);
		}
	}
}